src/sql/workbench/contrib/notebook/browser/models/notebookInput.ts

```diff
@@ -259,6 +258,10 @@ export abstract class NotebookInput extends EditorInput implements INotebookInpu
 		}
 	}
 
+	public get languageMode(): string | undefined {
```

Comment:
These shouldn't be returning undefined anymore now that you've removed the abstract text input so you can clean all that extra checking up.